ibm/service/contextbasedrestrictions/resource_ibm_cbr_zone.go

```diff
@@ -291,24 +305,58 @@ func resourceIBMCbrZoneCreate(context context.Context, d *schema.ResourceData, m
 	return resourceIBMCbrZoneRead(context, d, meta)
 }
 
+// waitForCbrZoneRead will leverage use retry.StateChangeConf due to the service's eventual consistency
+func waitForCbrZoneRead(cbrClient *contextbasedrestrictionsv1.ContextBasedRestrictionsV1, context context.Context, id string, timeout time.Duration) (interface{}, error) {
+	stateConf := &retry.StateChangeConf{
+		Pending: []string{cbrRuleReadPending},
+		Target:  []string{cbrRuleReadError, cbrRuleReadComplete, ""},
+		Refresh: func() (interface{}, string, error) {
+			log.Printf("[INFO] Retrying cbr rule (%s) read", id)
+			getZoneOptions := &contextbasedrestrictionsv1.GetZoneOptions{}
+			getZoneOptions.SetZoneID(id)
+			_, response, err := cbrClient.GetZoneWithContext(context, getZoneOptions)
+			if err != nil && response.StatusCode == 404 {
+				return response, cbrZoneReadPending, nil
+			} else if err != nil {
+				return response, cbrZoneReadError, err
+			} else {
+				return response, cbrZoneReadComplete, nil
+			}
+		},
+		Timeout:    timeout,
+		Delay:      20 * time.Second,
+		MinTimeout: 10 * time.Second,
+	}
+	return stateConf.WaitForStateContext(context)
+}
+
 func resourceIBMCbrZoneRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	contextBasedRestrictionsClient, err := meta.(conns.ClientSession).ContextBasedRestrictionsV1()
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
 	getZoneOptions := &contextbasedrestrictionsv1.GetZoneOptions{}
-
 	getZoneOptions.SetZoneID(d.Id())
 
 	zone, response, err := contextBasedRestrictionsClient.GetZoneWithContext(context, getZoneOptions)
 	if err != nil {
 		if response != nil && response.StatusCode == 404 {
-			d.SetId("")
-			return nil
+			//  Manual change. leverage retry for the read due to eventual consistency of the service.
```

Comment:
similar comment

Reply:
We can use retry.RetryContext instead of Resource.RetryContext  as sugested in deprecation